lib/models/environment-variables.coffee

```diff
@@ -229,9 +229,7 @@ getEnvironmentVariablesModel = (deps, opts) ->
 			return pine.get
```

Comment:
There's a few cases like this throughout the SDK, where we _always_ make the upfront request for the id, even though we might not need to. They exist because we want to consistently fail with a missing device error if the device doesn't exist. If we only make the single request below (to look up env vars for the device id) then we can't tell the difference between "the device doesn't exist" and "the device has no env vars". I'd like to avoid that in future, but unless there's an easy answer I'm planning to ignore it for now.

Reply:
How about expanding the `device_environment_variable`s through the `deviceModel().get` call?
That should throw the "the device doesn't exist" error properly? It might be an overkill though and I'm happy with the current implementation.

Reply:
Hmmm, yes, you're probably right that there's a route through there (though it may be messy). I don't want to worry about it right now, but I've filed an issue: https://github.com/resin-io/resin-sdk/issues/409